### ISSUE TITLE
VMM SDN Express: Logging broken

### ISSUE THREAD

**Issue by jrp2014** [Microsoft/SDN#182]
The VMM Express script LogWrite function

```
Function LogWrite
{
   Param (
		[string] $logstring
	)

	write-verbose $logstring
	Add-content $Logfile -value $logstring
}
```

takes a string argument, but in several places there are calls such as 

`LogWrite "COnnection String :" $ConnectionString`

Apart from the typos, the second argument ($ConnectionString in this case) is ignored.

**Comment by jrp2014** [comment]
To follow up to my own message, having tried to work with VMM SDN Express a bit further, it would save a great deal of angst if it had a bit more checking / environmental validation.  Part of the problem is (partly understandable) SCVMM state latency.  For example, the VMMExpress script gets as far as creating an NC, but then can't reach it, for reasons unspecified.  Rebooting the host VM cluster and retrying leads to a more intelligible message (that the NC requires a non-isolated network).  Why does the script not check first?  It would be slightly less troublesome if the original error message from SCVMM was helpful, but it isn't. Similarly, for uplink ports, net switch extensions, etc.  

All this slightly gives the impression that this script was written for a specific demo, rather than as a professional tool.

**Comment by ChristineBoersen** [comment]
Agreed would have saved me many hours trying to implement over a preexisting network (like many SMB do) if the scripts

- Checked the VM Network Isolation
- If the Logical Network isolation message was useful (you have to have the Logical Network in connected mode versus virtualized. You can tell which it is in by checking *BOTH* the IsLogicalNetworkDedfinitionIsolated: false and UseGRE: false
- If the Logical Switch was checked for MinimumBandwidthMode: Weight
- If the uplink port was checked for MinimumBandwidth set to 0 on all three values

Also if there was a way to offline the changes fopr some of these settings versus having to recreate
 - Logical Networks (To change from virtual to connected.. this one I can understand)
 - Logical Switches (to change VirtualSwitchExtensions or MinimumBandwitdhMode) 
 - Unassociate/Reassociate all the Hosts/VMs that used the LN/LS to the new ones (DOWNTIME!?!?!)
